lib/chainer/functions/connection/embed_id.rb

```diff
@@ -0,0 +1,50 @@
+module Chainer
+  module Functions
+    module Connection
+      class EmbedIDFunction < Chainer::Function
+        def initialize(ignore_label: nil)
+          @ignore_label = ignore_label
+        end
+
+        def self.embed_id(x, w, ignore_label: nil)
+          self.new(ignore_label: ignore_label).(x, w)
+        end
+
+        def forward_cpu(inputs)
+          (x, w) = inputs
+
+          unless @ignore_label
+            return [Chainer::Utils::Array.take(w, x.to_a, axis: 0)]
+          end
+
+          mask = Numo::Bit.cast(x.flatten.to_a.map{|i| i == @ignore_label}).reshape(*x.shape)
+          if mask.where.empty?
+            return [Chainer::Utils::Array.take(w, x.to_a, axis: 0)]
+          end
+          x = x.dup
+          x[mask.where] = 0
+          y = Chainer::Utils::Array.take(w, x.to_a, axis: 0).dup
+
+          ndindex = Chainer::Utils::Array.ndindex(y.shape.take(y.shape.size - 1))
+          mask.where.each {|i| y[*ndindex[i], true] = 0 }
```

Comment:
Avoid `each` by same reasons if possible.

Reply:
This section will not be reached except when @ignore_label is set to a value. 
Because it is a very unlikely event, I think that we do not have to deal with @ignore_label.
Do you think so?

Reply:
I am not sure of `ndindex`, I will look this later.